### PR TITLE
Make route related component fetch their own data

### PR DESCRIPTION
**Pull request by eleksi**
This change is [Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554)

### CODE REVIEW

File: ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetables.tsx [hunk 1]
```diff
@@ -1,17 +1,16 @@
-import { RouteWithJourneyPatternStopsFragment } from '../../../generated/graphql';
 import { VehicleRouteTimetableSection } from './VehicleRouteTimetableSection';
 
 interface Props {
-  routes: RouteWithJourneyPatternStopsFragment[];
+  routeIds: UUID[];
 }
 
-export const VehicleRouteTimetables = ({ routes }: Props): JSX.Element => {
+export const VehicleRouteTimetables = ({ routeIds }: Props): JSX.Element => {
```
Comment (pallost) [inline ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetables.tsx:7]: it's not necessarily in the scope of this PR, but this is a pretty bad name for these components. Would these be better?
VehicleRouteTimetables -> LineTimetables
VehicleRouteTimetable -> RouteTimetables

File: ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetableSection.tsx [hunk 2]
```diff
@@ -24,20 +24,29 @@ const testIds = {
...
 export const VehicleRouteTimetableSection = ({
-  route,
+  routeId,
   initiallyOpen = false,
 }: Props): JSX.Element => {
   const { t } = useTranslation();
   const [isOpen, toggleIsOpen] = useToggle(initiallyOpen);
   const { activeView, setShowPassingTimesByStop } = useTimetablesViewState();
 
+  const routeResult = useGetRouteDetailsByIdQuery({
```
Comment (pallost) [inline ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetableSection.tsx:34]: I'm pretty sure you don't need all the fields of the GetRouteDetailsById query here, please restrict it a bit!

File: ui/src/hooks/vehicle-service/useGetTimetables.ts [hunk 3]
```diff
@@ -199,7 +199,7 @@ export interface TimetableWithMetadata {
   journeyPatternId: UUID;
 }
 
-export const useGetTimetables = (journeyPatternId: UUID) => {
+export const useGetTimetables = (journeyPatternId?: UUID) => {
```
Comment (pallost) [inline ui/src/hooks/vehicle-service/useGetTimetables.ts:202]: this is a bad name for a hook. Is this rather a useGetRouteTimetables? (at least the parameter is a journey pattern id, that's why I think so)

File: ui/src/hooks/vehicle-service/useGetTimetables.ts [hunk 4]
```diff
@@ -208,6 +208,10 @@ export const useGetTimetables = (journeyPatternId: UUID) => {
   const [timetables, setTimetables] = useState<TimetableWithMetadata>();
 
   const getTimetablesForRoute = useCallback(async () => {
+    if (!journeyPatternId) {
+      return;
+    }
+
     const res = await getTimetablesForOperationDay({
```
Comment (pallost) [inline ui/src/hooks/vehicle-service/useGetTimetables.ts:215]: this is a bad name for this function... is this rather getRouteTimetablesForOperationDay?

**Review by pallost — CHANGES_REQUESTED** [review]

**Review by eleksi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554)* status: 0 of 14 files reviewed, 4 unresolved discussions (waiting on @pallost)

___
*[`ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetableSection.tsx` line 34 at r1](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554#-NQ_yINRSbyteM_jUdjQ-r1-34:-NQyjmzf7_o88yEnrkW6:bs6ocsw) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/7e3caa6fb732d6607cb8a360c4693000caabe82c/ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetableSection.tsx#L34)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

I'm pretty sure you don't need all the fields of the GetRouteDetailsById query here, please restrict it a bit!
</blockquote></details>

Good spot, infrastructure related data was not needed!
___
*[`ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetables.tsx` line 7 at r1](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554#-NQ_yINRSbyteM_jUdjR-r1-7:-NQyqkRKARCT14rW_aUm:bkzt7to) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/7e3caa6fb732d6607cb8a360c4693000caabe82c/ui/src/components/timetables/vehicle-schedule-details/VehicleRouteTimetables.tsx#L7)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

it's not necessarily in the scope of this PR, but this is a pretty bad name for these components. Would these be better?
VehicleRouteTimetables -> LineTimetables
VehicleRouteTimetable -> RouteTimetables
</blockquote></details>

We discussed this with Henkka and came up with these names. I hope they are little better than the original ones
___
*[`ui/src/hooks/vehicle-service/useGetTimetables.ts` line 202 at r1](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554#-NQ_yINSqxsfWz9nR9Bd-r1-202:-NQys5Wj1oDaVk3kVQff:b-896fix) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/7e3caa6fb732d6607cb8a360c4693000caabe82c/ui/src/hooks/vehicle-service/useGetTimetables.ts#L202)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

this is a bad name for a hook. Is this rather a useGetRouteTimetables? (at least the parameter is a journey pattern id, that's why I think so)
</blockquote></details>

Done.
___
*[`ui/src/hooks/vehicle-service/useGetTimetables.ts` line 215 at r1](https://reviewable.io/reviews/HSLdevcom/jore4-ui/554#-NQ_yINSqxsfWz9nR9Bd-r1-215:-NQys5oj1MZE8NSbDw7l:b-896fix) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/7e3caa6fb732d6607cb8a360c4693000caabe82c/ui/src/hooks/vehicle-service/useGetTimetables.ts#L215)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

this is a bad name for this function... is this rather getRouteTimetablesForOperationDay?
</blockquote></details>

Done.

**Pull request merged**